pkg/jobs/workers/sharings/sharing_updates.go

```diff
@@ -0,0 +1,178 @@
+package workers
+
+import (
+	"context"
+	"errors"
+	"strings"
+	"time"
+
+	"github.com/cozy/cozy-stack/client/auth"
+	"github.com/cozy/cozy-stack/pkg/consts"
+	"github.com/cozy/cozy-stack/pkg/couchdb"
+	"github.com/cozy/cozy-stack/pkg/couchdb/mango"
+	"github.com/cozy/cozy-stack/pkg/jobs"
+	"github.com/cozy/cozy-stack/pkg/permissions"
+)
+
+func init() {
+	jobs.AddWorker("sharingupdates", &jobs.WorkerConfig{
+		Concurrency:  4,
+		MaxExecCount: 3,
+		Timeout:      10 * time.Second,
+		WorkerFunc:   SharingUpdates,
+	})
+}
+
+var (
+	// ErrSharingIDNotUnique is used when several occurences of the same sharing id are found
+	ErrSharingIDNotUnique = errors.New("Several sharings with this id found")
+	// ErrSharingDoesNotExist is used when the given sharing does not exist.
+	ErrSharingDoesNotExist = errors.New("Sharing does not exist")
+	// ErrDocumentNotLegitimate is used when a shared document is triggered but
+	// not legitimate for this sharing
+	ErrDocumentNotLegitimate = errors.New("Triggered illegitimate shared document")
+	//ErrRecipientDoesNotExist is used when the given recipient does not exist
+	ErrRecipientDoesNotExist = errors.New("Recipient with given ID does not exist")
+	// ErrRecipientHasNoURL is used to signal that a recipient has no URL.
+	ErrRecipientHasNoURL = errors.New("Recipient has no URL")
+)
+
+// TriggerEvent describes the fields retrieved after a triggered event
+type TriggerEvent struct {
+	Event   *EventDoc       `json:"event"`
+	Message *SharingMessage `json:"message"`
+}
+
+// EventDoc describes the event returned by the trigger
+type EventDoc struct {
+	Type string `json:"type"`
+	Doc  *couchdb.JSONDoc
+}
+
+// SharingMessage describes a sharing message
+type SharingMessage struct {
+	SharingID string `json:"sharing_id"`
+	DocType   string `json:"doctype"`
+}
+
+// Sharing describes the sharing document structure
+type Sharing struct {
+	SharingType      string             `json:"sharing_type"`
+	Permissions      permissions.Set    `json:"permissions,omitempty"`
+	RecipientsStatus []*RecipientStatus `json:"recipients,omitempty"`
+}
+
+// RecipientStatus contains the information about a recipient for a sharing
+type RecipientStatus struct {
+	Status       string               `json:"status,omitempty"`
+	RefRecipient couchdb.DocReference `json:"recipient,omitempty"`
+	AccessToken  *auth.AccessToken
+}
+
+// SharingUpdates handles shared document updates
+func SharingUpdates(ctx context.Context, m *jobs.Message) error {
+	domain := ctx.Value(jobs.ContextDomainKey).(string)
+
+	event := &TriggerEvent{}
+	err := m.Unmarshal(&event)
+	if err != nil {
+		return err
+	}
+	sharingID := event.Message.SharingID
+	docType := event.Message.DocType
+	docID := event.Event.Doc.M["_id"].(string)
+
+	// Get the sharing document
+	db := couchdb.SimpleDatabasePrefix(domain)
```

Comment:
All right, `SimpleDatabasePrefix` was also used in `share_data.go`, so I changed both and fixed the tests accordingly.